packages/server/app/Controllers/Http/AuthenticationController.ts

```diff
@@ -68,9 +72,25 @@ export default class AuthenticationController {
             password,
         });
 
+        const emailConfirmationTokenType = await TokenType.findByOrFail(
+            'name',
+            TokenTypeName.enum.EMAIL_CONFIRMATION,
+        );
+        const confirmationTokenValue = await Token.generateCode();
```

Comment:
thoughts: I was thinking, but why would generate the code, you could just save the token and retrieve after all. NO you can't, it's hashed ..

Reply:
Haha, I thought the same thing!